### PR TITLE
(fix): Be more permissive during retries in workflow remote execution initialization

### PR DESCRIPTION
**Pull request by justinkaseman**

### CODE REVIEW

File: core/capabilities/remote/executable/client.go [hunk 1]
```diff
@@ -208,10 +208,11 @@ func (c *client) sendRequest(req *request.ClientRequest) error {
 	c.lggr.Debugw("executing remote execute capability", "requestID", req.ID())
 
 	if _, ok := c.requestIDToCallerRequest[req.ID()]; ok {
-		return fmt.Errorf("request for ID %s already exists", req.ID())
+		c.lggr.Debugw("request for ID already exists", "requestID", req.ID())
```
Comment (krehermann) [inline core/capabilities/remote/executable/client.go:211]: imo easier to read if log and return and fall thru

```
if _, exists := ....; exists {
log.Debug
return nil
}
c.foo = bar
```

**Review by krehermann — DISMISSED** [review]

File: core/capabilities/remote/executable/client.go (hunk 1 above)
Comment (justinkaseman) [reply core/capabilities/remote/executable/client.go:211]: Fair. I don't have a strong opinion, so I'll change it

**Review by krehermann — DISMISSED** [review]

**Review by bolekk — CHANGES_REQUESTED** [review]
I'm afraid this is not enough :(

See my Slack message and the proposed alternative fix: https://github.com/smartcontractkit/chainlink/pull/15352/files

**Comment by justinkaseman** [comment]
Closing for @ettec's larger rework

**Pull request closed without merging**